crates/flux-middle/src/sort_of.rs

```diff
@@ -41,6 +43,22 @@ impl GlobalEnv<'_, '_> {
         if ty.is_integral() { Ok(Some(rty::Sort::Int)) } else { self.sort_of_rust_ty(def_id, ty) }
     }
 
+    pub fn sort_of_spec_func(self, spec_func: &SpecFuncKind) -> rty::PolyFuncSort {
+        match spec_func {
+            SpecFuncKind::Def(name) | SpecFuncKind::Uif(name) => self.func_sort(name),
+            SpecFuncKind::Thy(itf) => THEORY_FUNCS.get(itf).unwrap().sort.clone(),
+            SpecFuncKind::Cast => {
+                rty::PolyFuncSort::new(
+                    List::from_arr([SortParamKind::Sort, SortParamKind::Sort]),
+                    rty::FuncSort::new(
+                        vec![rty::Sort::Var(rty::ParamSort::from(0_usize))],
+                        rty::Sort::Var(rty::ParamSort::from(1_usize)),
+                    ),
+                )
+            }
+        }
+    }
+
```

Comment:
Please revert this change. At some point, I want to stop treating theory functions as "functions" and have dedicated rules for them, so we can get rid of parametric bit vector sizes. This change goes in the wrong direction.